src/rolch/online_gamlss.py

```diff
@@ -938,20 +880,14 @@ def _inner_update(
 
     def predict(
         self,
-        x0: np.ndarray,
-        x1: Optional[np.ndarray] = None,
-        x2: Optional[np.ndarray] = None,
-        x3: Optional[np.ndarray] = None,
+        X: np.ndarray,
         what: str = "response",
         return_contributions: bool = False,
     ) -> np.ndarray:
         """Predict the distibution parameters given input data.
 
         Args:
-            x0 (np.ndarray): Design matrix for the 1st distribution parameter $X_\\mu$
-            x1 (Optional[np.ndarray], optional): Design matrix for the 2nd distribution parameter $X_\\sigma$. Defaults to None.
-            x2 (Optional[np.ndarray], optional): Design matrix for the 3rd distribution parameter $X_\\nu$. Defaults to None.
-            x3 (Optional[np.ndarray], optional): Design matrix for the 4th distribution parameter $X_\\tau$. Defaults to None.
+            X (np.ndarray): Design matrix.
             what (str, optional): Predict the response or the link. Defaults to "response".
```

Comment:
Hm, I do not quite get the API design from this, as the `what` argument - important - is incompletely documented, similar the return. I would strongly advise to make this as precise as possible, and explain the return object for each possible value of `what` clearly and extensively. After all, besides the more "obvious" `update` point, this is the second key departure from the sklearn API.

Reply:
GAMLSS models $g(\theta) = \eta = X\beta_\theta$ where $g(\cdot)$ is the link function that ensures that your parameter is in the correct domain. For prediction we can either return $X\beta_\theta$ or $g^{-1}(X\beta)$ - and that is what the `what` parameter regulates. 

My guess that in 90% of the cases you want the $g^{-1}(X\beta_\theta)$, i.e the parameter estimates and not the link function. The current API is more aligned with `R`'s `gamlss::gamlss.predict(model, newdata, type="link")`, but I'm not so super sure about the best design here. 

We could also have 
- `Estimator.predict()` = Mean (or location) prediction 
- `Estimator.predict_distribution()` = Prediction of distribution parameters $g^{-1}(X\beta_\theta)$
- `Estimator.predict_distribution_link()` = Prediction of linked distribution parameters $X\beta_\theta$
- ... more methods for intervals etc

Some of these generalize quite well to other distributional regression methods, other don't. 

@BerriJ what do you think?

Reply:
I have to think about this a little more. My first intuition is in favour of the `what` argument. I guess, most users won't ever use this as they usually want to obtain  $g^{-1}(X\beta)$.